### PR TITLE
Delete unused new question route

### LINKED ISSUE #19677: Remove `/question/new` route
Once we decided to update and simplify starting points for asking the question, we changed that flow. Please see: https://github.com/metabase/metabase/pull/19572

In that process the route `/question/new` was forgotten. It still leads to the well known page that lets one choose between Simple Question, Custom Question and a Native Query
image

Unless someone manually enters that route, it's not possible anymore to reach it.
Product decided to remove it.

### Gotchas:
- ~~Large number of e2e tests is still relying on this flow and should be updated accordingly~~ (done in #21558 and #21564)

### PR DESCRIPTION
**Pull request by gusaiani**
Closes #19677

Here we clean up the route and page where we used to display a question type picker (simple question / custom question / native query).

We update code for the model equivalent so that it reflects the fact that this directory is now all about models.

Also, we s/dataset/model everywhere we touch here to converge our codebase towards calling models models.

### How to Test

Go to http://localhost:3000/question/new

You should no longer see the question type picker, instead land on the data picker for a custom question.

The app hasn't had a link to `/question/new` since these links are now inside `+ New`.

### CODE REVIEW

File: frontend/src/metabase/nav/containers/MainNavbar/MainNavbarContainer/MainNavbarContainer.tsx
Comment (kulyk) [inline frontend/src/metabase/nav/containers/MainNavbar/MainNavbarContainer/MainNavbarContainer.tsx:26]: Just stepping by and figured it'd be more helpful to ping you earlier than later. Please use "model" instead of "dataset" for the new code. All the existing code using "dataset" was introduced with a WIP name in mind, and we'd need to spend some time renaming all that eventually; but for new code please use "model"

Comment (gusaiani) [reply frontend/src/metabase/nav/containers/MainNavbar/MainNavbarContainer/MainNavbarContainer.tsx]: Cool @kulyk, yes, great that we're choosing one and going with it 👍

File: frontend/src/metabase/routes.jsx
Comment (kulyk) [inline frontend/src/metabase/routes.jsx:44]: Can we use the `NewModelOptions` name here?

File: frontend/src/metabase/new_model/containers/NewModelOptions.jsx
Comment (kulyk) [inline frontend/src/metabase/new_model/containers/NewModelOptions.jsx:19]: Should we use the existing `getHasDataAccess` and `getHasNativeWrite` defined in `metabase/nav/selectors` instead of duplicating them? Maybe it'd be worth extracting them somewhere else instead of `metabase/nav` as they're pretty helpful outside the nav too

**Review by kulyk — COMMENTED** [review]
Thank you! I think we can now also remove [this URL helper](https://github.com/metabase/metabase/blob/2a9fcbf3a74702ecd2c327f8a41354e24a873c10/frontend/src/metabase/lib/urls/questions.ts#L19)? Also there are two occurrences of `/quesiton/new` in our cross-version test suite (check out [this line](https://github.com/metabase/metabase/blob/2a9fcbf3a74702ecd2c327f8a41354e24a873c10/frontend/test/metabase/scenarios/cross-version/source/03-questions.cy.spec.js#L6))

**Comment by kulyk** [comment]
Adding @nemanjaglumac to reviewers — AFAIR this route was used quite heavily in E2E tests, so it'd be very helpful if you can take a look at this change Nemanja

**Comment by gusaiani** [comment]
@kulyk @nemanjaglumac 

Thanks Anton.

Yes, to my surprise, the route was not used at all at least in latest master (meaning e2e tests too), because no tests needed to be changed.

Nemanja's eyes are always welcome.

File: frontend/src/metabase/new_model/containers/NewModelOptions.jsx
Comment (gusaiani) [reply frontend/src/metabase/new_model/containers/NewModelOptions.jsx:19]: Yes, that makes sense to me as well. 

Any idea of a place off the top of your head, @kulyk ?

Comment (kulyk) [reply frontend/src/metabase/new_model/containers/NewModelOptions.jsx:19]: Maybe add a new file to `metabase/selectors`? E.g. `metabase/selectors/data`

File: frontend/src/metabase/routes.jsx
Comment (gusaiani) [reply frontend/src/metabase/routes.jsx]: Updated.

File: frontend/src/metabase/new_model/containers/NewModelOptions.jsx
Comment (gusaiani) [reply frontend/src/metabase/new_model/containers/NewModelOptions.jsx:19]: Yep.

I thought of adding these to `metabase/selectors/user` as it's the user's access to data.

Or calling it `metabase/selectors/data-acess` or something.

Went with `metabase/selectors/data`, thank you.

Comment (kulyk) [reply frontend/src/metabase/new_model/containers/NewModelOptions.jsx:19]: @gusaiani, should we use `metabase/selectors/data` in the new model flow as well? So we can remove `metabase/new_model/selectors`

Comment (gusaiani) [reply frontend/src/metabase/new_model/containers/NewModelOptions.jsx]: Absolutely. Updated.

**Review by kulyk — APPROVED** [review]
Nice work! 🎉

**Comment by nemanjaglumac** [comment]
This is why it doesn't exist in E2E tests anymore :)
- https://github.com/metabase/metabase/pull/21564
- https://github.com/metabase/metabase/pull/21558

However, if you merge this, a weekly cross-version tests will fail.
That is, they will if I don't fix them before Sunday. That shouldn't stop you from merging.

There's a valid concern that @flamber raised on many occasions - there is potentially a large number of users who are still using this route in their own automation scripts. If we do this, we need to officially deprecate the route and be public about it.

**Comment by gusaiani** [comment]
@nemanjaglumac, thank you very much.

I'm continuing to rerun CI until we get past fails.
So I won't merge until tomorrow (Tuesday).

Once we merge, how do we go about officially deprecating the route and being public about it? 
Happy to be part of this so we make sure this is done nicely.

**Comment by nemanjaglumac** [comment]
@gusaiani should this be part of 45 or the 46 release?

**Comment by gusaiani** [comment]
@nemanjaglumac Could well be 46.

**Pull request merged**
